### PR TITLE
Enforced non page posts only being returned by /posts endpoint in Content API

### PR DESCRIPTION
**Pull request by naz**
refs #9866

- Added logic ensuring page filter is always set to false in posts endpoint
- Added functional tests to pages and posts
- Added absolute_url logic in pages controller

### CODE REVIEW

File: core/server/api/v2/utils/serializers/input/posts.js [hunk 1]
```diff
@@ -1,7 +1,30 @@
 const _ = require('lodash');
+const debug = require('ghost-ignition').debug('api:v2:utils:serializers:posts:pages');
```
Comment (kirrg001) [inline core/server/api/v2/utils/serializers/input/posts.js]: > :serializers:posts:pages

typo 🌶

File: core/server/web/api/v2/content/routes.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ module.exports = function apiRoutes() {
     router.get('/configuration', api.http(api.configuration.read));
 
     // ## Posts
-    router.get('/posts', mw.authenticatePublic, api.http(api.posts.browse));
-    router.get('/posts/:id', mw.authenticatePublic, api.http(api.posts.read));
+    router.get('/posts', mw.authenticatePublic, apiv2.http(apiv2.posts.browse));
+    router.get('/posts/:id', mw.authenticatePublic, apiv2.http(apiv2.posts.read));
     router.get('/posts/slug/:slug', mw.authenticatePublic, api.http(api.posts.read));
```
Comment (kirrg001) [inline core/server/web/api/v2/content/routes.js]: What about this endpoint?:)

File: core/server/api/v2/utils/serializers/input/posts.js (hunk 1 above)
Comment (gargol) [reply core/server/api/v2/utils/serializers/input/posts.js]: ah damn it :)

File: core/test/functional/api/v0.1/public_api_spec.js [hunk 3]
```diff
@@ -77,29 +77,6 @@ describe('Public API', function () {
             });
     });
 
-    it('browse pages', function (done) {
```
Comment (kirrg001) [inline core/test/functional/api/v0.1/public_api_spec.js]: 👻

File: core/server/api/v2/utils/serializers/input/posts.js [hunk 4]
```diff
@@ -1,7 +1,30 @@
 const _ = require('lodash');
+const debug = require('ghost-ignition').debug('api:v2:utils:serializers:input:posts');
 
 module.exports = {
+    all(apiConfig, frame) {
+        debug('all');
+
+        // CASE: the content api endpoints for posts should only return non page type resources
+        if (frame.options.filter) {
```
[the remark this answers is not in the document]
Comment (kirrg001) [reply core/server/api/v2/utils/serializers/input/posts.js]: yeah 👍 only the content api differentiates between posts and pages

[the remark this answers is not in the document]
Comment (kirrg001) [reply core/server/api/v2/utils/serializers/input/posts.js]: Just need to ask for `frame.options.context.public` 👍

[the remark this answers is not in the document]
Comment (kirrg001) [reply core/server/api/v2/utils/serializers/input/posts.js]: Discussed: permission stage happens currently after serialisation. Might change in the future.
We have access to context.user and context.api_key and context.client_id (content key auth is not merged into master yet)

File: core/test/functional/api/v2/content/posts_spec.js [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+                jsonResponse.meta.pagination.pages.should.eql(1);
+                jsonResponse.meta.pagination.total.should.eql(0);
+                should.equal(jsonResponse.meta.pagination.next, null);
+                should.equal(jsonResponse.meta.pagination.prev, null);
+
+                done();
+            });
+    });
+
+    it('browse posts: request absolute urls', function (done) {
```
Comment (kirrg001) [inline core/test/functional/api/v2/content/posts_spec.js]: can be merged with the first test

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**